test/runtests.jl

```diff
@@ -153,6 +153,25 @@ const url = "http://127.0.0.1:$port"
             wait(t)  # handle errors
         end
     end
+    
+    @testset "Type inference profiling" begin
+        @testset "typeinf start/stop endpoints" begin
+            resp = HTTP.get("$url/typeinf_profile_start", retry=false, status_exception=false)
+            @test resp.status == 200
+            @test String(resp.body) == "Type inference profiling started."
+            
+            # workload
+            @eval foo() = 2
+            @eval foo()
+            
+            resp = HTTP.get("$url/typeinf_profile_stop", retry=false, status_exception=false)
+            @test resp.status == 200
+            data = read(IOBuffer(resp.body), String)
+            # Test that there's something here
+            # TODO: actually parse the profile
+            @test length(data) > 100
```

Comment:
if we changed this to return the original profile data, not the pprof flamegraph, it might be easier to test as well, since `foo` should be one of the top-level inference invocations.